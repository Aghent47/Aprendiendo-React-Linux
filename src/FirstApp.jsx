
// import React, { Fragment } from 'react' = <> 
import PropTypes from 'prop-types';

const getNameUsers = () =>  {
  const users = {
    name: 'Neider Hernández',
    age: '25',
    city: 'Ciudad de México',
   }
  return users;  
}

// Desestructuando los Props {title, subTitle, age}
export const FirstApp = ({ title, subTitle, name}) => { 
 
  return (
    <>
    <h1>{ title }</h1>
    <p> { name } </p>
    {/* <h2> {`${getNameUsers().name} ${age} Años `}</h2>
    <h3> { subtitle } </h3> */}
    <p> { subTitle } </p>
    
    </>
  )
  
}

// Definiendo los PropTypes

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string,
  name: PropTypes.string, 
}

// Definiendo los defaultProps (Se ejecutan primero que los PropTypes)
FirstApp.defaultProps = {
  name: 'Neider Hernández',
  /* title: "Titulo no especificado", */
  subTitle: "Software Engineer",

}