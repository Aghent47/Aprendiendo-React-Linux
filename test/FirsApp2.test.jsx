import { render, screen } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";

describe('Pruebas Component <FirstApp/>', () => {

    const title = 'Hola, soy Neider Herández';
    const subTitle = 'Software Engineer';

    test('debe de hacer match con el snapshot', () => { 
        
        const { container } = render(< FirstApp title={ title } />)
        
        expect(container).toMatchSnapshot();
    });

    test('debe de mostrar el mensaje "Hola, soy Neider Hernández"', () => { 

        render(< FirstApp title={ title }/>)
        /* screen.debug();  */// obtenemos la ultima version de un componente renderizado
        expect(screen.getByText(title)).toBeTruthy();
     });

     test('debe de mostrar el titulo en un <h1></h1>', () => { 

        render(< FirstApp title={ title }/>)
        expect(screen.getByRole('heading', {level: 1}).innerHTML).toContain(title)

      });

      test('debe mosreae el subtitulo enviado por la props', () => { 

        render(<FirstApp title={ title }
        subTitle={subTitle}
        />)

        expect( screen.getAllByText(subTitle).length ).toBe(2);

       })
    
});