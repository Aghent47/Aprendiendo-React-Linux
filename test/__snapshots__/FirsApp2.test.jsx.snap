// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas Component <FirstApp/> debe de hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola, soy Neider Herández
  </h1>
  <p>
     
    Neider Hernández
     
  </p>
  <p>
     
    Software Engineer
     
  </p>
  <p>
     
    Software Engineer
     
  </p>
</div>
`;
